dhis-2/dhis-services/dhis-service-reporting/src/test/java/org/hisp/dhis/orgunitprofile/OrgUnitProfileServiceTest.java

```diff
@@ -0,0 +1,204 @@
+    {
+        OrgUnitProfile orgUnitProfile = new OrgUnitProfile();
+        orgUnitProfile.getAttributes().add( "Attribute1" );
+        orgUnitProfile.getAttributes().add( "Attribute2" );
+        orgUnitProfile.getDataItems().add( "DataItem1" );
+        orgUnitProfile.getDataItems().add( "DataItem2" );
+        orgUnitProfile.getGroupSets().add( "GroupSet1" );
+        orgUnitProfile.getGroupSets().add( "GroupSet2" );
+        service.saveOrgUnitProfile( orgUnitProfile );
+
+        OrgUnitProfile savedProfile = service.getOrgUnitProfile();
+        assertEquals( 2, savedProfile.getAttributes().size() );
+        assertEquals( 2, savedProfile.getDataItems().size() );
+        assertEquals( 2, savedProfile.getGroupSets().size() );
+        assertTrue( savedProfile.getAttributes().contains( "Attribute1" ) );
+        assertTrue( savedProfile.getDataItems().contains( "DataItem2" ) );
+        assertTrue( savedProfile.getGroupSets().contains( "GroupSet1" ) );
+    }
+
+    @Test
+    public void testGetProfileData()
+    {
+        Attribute attribute = createAttribute( 'A' );
+        attribute.setOrganisationUnitAttribute( true );
+        manager.save( attribute );
+
+        OrganisationUnit orgUnit = createOrganisationUnit( "A" );
+        orgUnit.getAttributeValues().add( new AttributeValue( "testAttributeValue", attribute ) );
+        manager.save( orgUnit );
+
+        OrganisationUnitGroup group = createOrganisationUnitGroup( 'A' );
+        group.addOrganisationUnit( orgUnit );
+        manager.save( group );
+
+        OrganisationUnitGroupSet groupSet = createOrganisationUnitGroupSet( 'A' );
+        groupSet.addOrganisationUnitGroup( group );
+        manager.save( groupSet );
+
+        DataElement dataElement = createDataElement( 'A' );
+        manager.save( dataElement );
+
+        Period period = createPeriod( "202106" );
+        manager.save( period );
+
+        OrgUnitProfile orgUnitProfile = new OrgUnitProfile();
+        orgUnitProfile.getAttributes().add( attribute.getUid() );
+        orgUnitProfile.getDataItems().add( dataElement.getUid() );
+        orgUnitProfile.getGroupSets().add( groupSet.getUid() );
+        service.saveOrgUnitProfile( orgUnitProfile );
+
+        // Mock analytic query for data value
+        Map<String, Object> mapDataItem = new HashMap<>();
+        mapDataItem.put( dataElement.getUid(), "testDataValue" );
+        Mockito.when( mockAnalyticsService.getAggregatedDataValueMapping( any( DataQueryParams.class ) ) )
+            .thenReturn( mapDataItem );
+
+        OrgUnitProfileData data = mockService.getOrgUnitProfileData( orgUnit.getUid(), period.getIsoDate() );
+
+        assertEquals( "testAttributeValue", data.getAttributes().get( 0 ).getValue() );
+        assertEquals( "testDataValue", data.getDataItems().get( 0 ).getValue() );
+        assertEquals( group.getDisplayName(), data.getGroupSets().get( 0 ).getValue() );
+    }
+
+    @Test
+    public void testValidator()
```

Comment:
Add a test for retrieving the org unit profile, which verifies the behavior when there is no existing org unit profile, and when there is an existing profile.

Reply:
fixed